app/jobs/algolia_reindex_job.rb

```diff
@@ -15,8 +15,28 @@
 class AlgoliaReindexJob
   def perform
     Resource.where(status: :approved).reindex
+
+    # Batch synonyms, with replica forwarding and atomic replacement of existing synonyms
+    forward_to_replicas = true
+    replace_existing_synonyms = true
+    Resource.index.batch_synonyms(format_synonyms_list, forward_to_replicas, replace_existing_synonyms)
+
     # Since Service and Resource share an algolia index, we use `reindex!` so
     # Service reindexing does not clear out Resource index records.
     Service.where(status: :approved).reindex!
   end
+
+  private
+
+  def format_synonyms_list
+    synonyms_list = []
+    SynonymGroup.all.each do |group|
```

Comment:
I _think_ it might be better memory-wise to use `find_each` here because `all` will first get all the records for the model _then_ `each` will iterate through, whereas `find_each` will query and iterate in batches. Better explanation here: https://rubyinrails.com/2017/11/16/use-find-each-instead-of-all-each-in-rails/